Moderator/admin control endpoints

jagerman opened the pull request:
Adds endpoints for adding/removing moderators/admins from rooms/server.

jagerman committed:
api doc formatting fix

jagerman committed:
Drop banned request earlier

No point in wasting cycles doing auth validation if we're just going to
drop the request anyway because the pubkey is banned.

jagerman committed:
Implement moderator/admin endpoints

Allows adding/removing room-specific and global moderators/admins.

jagerman committed:
Simplify making signed requests in test suite

jagerman committed:
api.yaml syntax fixes

majestrate commented on sogs/routes/users.py:
```diff
@@ -0,0 +1,115 @@
+        )
+        abort(http.BAD_REQUEST)
+    elif (admin, mod) == (True, False):
+        app.logger.warning("Invalid moderator call: admin=True, moderator=False is impossible")
```
this should be error level

jagerman replied:
Leaving as warning because this is a client-side error, but still wanted to log it in case a server admin is trying to figure out why some (wrong) client call "doesn't work".

jagerman committed:
Relax test timing

On the RPi workers this sometimes fails (i.e. taking more than a second)
because of the large number of posts we insert in this test code. Relax
to 10s.
